scripts/verify-build-artifacts.js

```diff
@@ -0,0 +1,133 @@
+governing permissions and limitations under the License.
+*/
+
+import fs from 'fs';
+import path from 'path';
+import glob from 'fast-glob';
+import 'colors';
+
+function verifyCustomElementsJson() {
+    // Components that don't need their own custom-elements.json manifest
+    const customElementsIgnoreList = [
+        'packages/modal',
+        'packages/iconset',
+        'packages/clear-button',
+        'packages/close-button',
+    ];
+
+    const packages = glob.sync('packages/*/', { onlyDirectories: true });
+    for (const pkg of packages) {
+        // Remove trailing slash for comparison
+        const pkgPath = pkg.replace(/\/$/, '');
+        if (customElementsIgnoreList.includes(pkgPath)) {
+            continue;
+        }
+        const customElementsPath = path.join(pkg, 'custom-elements.json');
+        if (!fs.existsSync(customElementsPath)) {
+            throw new Error(`Missing custom-elements.json in ${pkg}`);
+        }
+    }
+}
+
+function verifyVersionJs() {
+    const basePackageJson = JSON.parse(
+        fs.readFileSync('tools/base/package.json', 'utf8')
+    );
+    const versionJsPath = 'tools/base/src/version.js';
+
+    if (!fs.existsSync(versionJsPath)) {
+        throw new Error('version.js file is missing');
+    }
+
+    const versionContent = fs.readFileSync(versionJsPath, 'utf8');
+    const versionMatch = versionContent.match(/version = ['"]([^'"]+)['"]/);
+
+    if (!versionMatch) {
+        throw new Error('Could not find version in version.js');
+    }
+
+    const versionJs = versionMatch[1];
+    if (versionJs !== basePackageJson.version) {
+        throw new Error(
+            `Version mismatch: version.js (${versionJs}) does not match tools/base/package.json (${basePackageJson.version})`
+        );
+    }
+}
+
+function verifyBuildArtifacts() {
+    const packages = glob.sync('packages/*/', { onlyDirectories: true });
+
+    const requiredFilesIgnoreList = [
+        'packages/clear-button', // extends button
+        'packages/close-button', // extends button
+        'packages/search-button', // extends button
+        'packages/icons-ui', // extends icon
+        'packages/icons-workflow', // extends icon
+        'packages/iconset', // extends icon
+        'packages/modal', // extends dialog
+    ];
+
+    // Required files for each package
+    const requiredFiles = [
+        ['src/index.js', 'main compiled JS file'],
+        ['src/index.dev.js', 'dev build file'],
+        ['src/*.d.ts', 'TypeScript declaration files'],
+        ['sp-*.js', 'component definition file'],
+        ['sp-*.dev.js', 'dev component definition file'],
+        ['sp-*.d.ts', 'component definition type file'],
+    ];
+
+    for (const pkg of packages) {
+        const pkgPath = pkg.replace(/\/$/, '');
+
+        // First verify src directory exists
+        const srcPath = path.join(pkg, 'src');
+        if (!fs.existsSync(srcPath)) {
```

Comment:
fs.existsSync(srcPath) only checks if src/ exists. We should also verify it's not empty.

Reply:
It should be covered below when we check for:
```
['src/index.js', 'main compiled JS file'],
['src/index.dev.js', 'dev build file'],
['src/*.d.ts', 'TypeScript declaration files'],
```
but I added an extra check just to be sure.